idaes/generic_models/properties/core/examples/tests/test_CO2_bmimPF6_PR.py

```diff
@@ -0,0 +1,318 @@
+        for p in model.params.phase_list:
+            for j in model.params.component_list:
+                if j == "bmimPF6":
+                    assert model.props[1].get_material_density_terms("Liq", j) == (
+                        model.props[1].dens_mol_phase["Liq"] *
+                        model.props[1].mole_frac_phase_comp["Liq", j])
+                else:
+                    assert model.props[1].get_material_density_terms(p, j) == (
+                        model.props[1].dens_mol_phase[p] *
+                        model.props[1].mole_frac_phase_comp[p, j])
+
+    @pytest.mark.unit
+    def test_get_energy_density_terms(self, model):
+        for p in model.params.phase_list:
+            assert model.props[1].get_energy_density_terms(p) == (
+                model.props[1].dens_mol_phase[p] *
+                model.props[1].enth_mol_phase[p])
+
+    @pytest.mark.unit
+    def test_default_material_balance_type(self, model):
+        assert model.props[1].default_material_balance_type() == \
+            MaterialBalanceType.componentTotal
+
+    @pytest.mark.unit
+    def test_default_energy_balance_type(self, model):
+        assert model.props[1].default_energy_balance_type() == \
+            EnergyBalanceType.enthalpyTotal
+
+    @pytest.mark.unit
+    def test_get_material_flow_basis(self, model):
+        assert model.props[1].get_material_flow_basis() == \
+            MaterialFlowBasis.molar
+
+    @pytest.mark.unit
+    def test_define_state_vars(self, model):
+        sv = model.props[1].define_state_vars()
+
+        assert len(sv) == 4
+        for i in sv:
+            assert i in ["flow_mol",
+                         "mole_frac_comp",
+                         "temperature",
+                         "pressure"]
+
+    @pytest.mark.unit
+    def test_define_port_members(self, model):
+        sv = model.props[1].define_state_vars()
+
+        assert len(sv) == 4
+        for i in sv:
+            assert i in ["flow_mol",
+                         "mole_frac_comp",
+                         "temperature",
+                         "pressure"]
+
+    @pytest.mark.unit
+    def test_define_display_vars(self, model):
+        sv = model.props[1].define_display_vars()
+
+        assert len(sv) == 4
+        for i in sv:
+            assert i in ["Total Molar Flowrate",
+                         "Total Mole Fraction",
+                         "Temperature",
+                         "Pressure"]
+
+    @pytest.mark.initialize
+    @pytest.mark.solver
+    @pytest.mark.skipif(solver is None, reason="Solver not available")
+    @pytest.mark.component
+    def test_initialize(self, model):
+        # Fix state
+        model.props[1].flow_mol.fix(1)
+        model.props[1].temperature.fix(200.00)
+        model.props[1].pressure.fix(101325)
+        model.props[1].mole_frac_comp["carbon_dioxide"].fix(1/2)
+        model.props[1].mole_frac_comp["bmimPF6"].fix(1/2)
+
+        assert degrees_of_freedom(model.props[1]) == 0
```

Comment:
Have a look at the `initialization_tester` method in `idaes/core/util/testing` - it automates all of this and more.